nxc/modules/snipped.py

```diff
@@ -0,0 +1,126 @@
+import ntpath
+import os
+from impacket import smb, smb3
+from os.path import join, exists
+from dploot.lib.smb import DPLootSMBConnection
+from dploot.lib.target import Target  
+from nxc.paths import NXC_PATH
+
+class NXCModule:
+
+    name = "snipped"
+    description = "Downloads screenshots taken by the (new) Snipping Tool."
+    supported_protocols = ["smb"]
+    opsec_safe = True
+    multiple_hosts = True
+
+    def __init__(self):
+        self.context = None
+        self.module_options = None
+
+    def options(self, context, module_options):
+        """
+        USERS           Download only specified user(s); format: -o USERS=user1,user2,user3
+        """
+        self.context = context
+        self.screenshot_path_stub = r"Pictures\Screenshots"
+        self.users = module_options["USERS"].split(",") if "USERS" in module_options else None
+
+    def on_admin_login(self, context, connection):
+        self.context = context
+        self.connection = connection
+        self.share = "C$"
+
+        self.hostname = connection.hostname
+        self.domain = connection.domain
+        self.username = connection.username
+        self.password = getattr(connection, "password", "")
+        self.host = connection.host
+        self.kerberos = connection.kerberos
+        self.lmhash = getattr(connection, "lmhash", "")
+        self.nthash = getattr(connection, "nthash", "")
+        self.aesKey = connection.aesKey
+        self.use_kcache = getattr(connection, "use_kcache", False)
+
+        target = Target.create(
+            domain=self.domain,
+            username=self.username,
+            password=self.password,
+            target=self.hostname + "." + self.domain if self.kerberos else self.host,
+            lmhash=self.lmhash,
+            nthash=self.nthash,
+            do_kerberos=self.kerberos,
+            aesKey=self.aesKey,
+            no_pass=True,
+            use_kcache=self.use_kcache,
+        )
+
+        dploot_conn = self.upgrade_connection(target=target, connection=connection.conn)
+
+        output_base_dir = join(NXC_PATH, "modules", "snipped", "screenshots")
+        os.makedirs(output_base_dir, exist_ok=True)
+
+        context.log.debug("Getting all user folders")
+        try:
+            user_folders = dploot_conn.listPath(self.share, "\\Users\\*")
+        except Exception as e:
+            context.log.fail(f"Failed to list user folders: {e}")
+            return
+
+        context.log.debug(f"User folders: {user_folders}")
+        if not user_folders:
+            context.log.fail("No User folders found!")
+            return
+        else:
+            context.log.display("Attempting to download screenshots if existent.")
```

Comment:
As this is a module that would probably be run against several hosts at once i would prefer having this as and `.info` log message.

Reply:
Done:

https://github.com/Yeeb1/NetExec/blob/af512f1e6ba7dd37f11e430bdf18b0d22347ddd7/nxc/modules/snipped.py#L47